### ISSUE TITLE
Fix "zero dates" handling for cftime > 1.0.1

### ISSUE THREAD

**Issue by pp-mo** [SciTools/iris#3248]
Problems are well covered elsewhere:
  * #3215 original occurrence
  * #3239 reoccurrence
  * #3245 withdrawn proposal + hints of a better way

Discussed [here](https://github.com/SciTools/iris/issues/3239#issuecomment-452314524)

**Comment by rcomer** [comment]
A I right in thinking this was fixed by #3259?

**Comment by pp-mo** [comment]
> Am I right in thinking this was fixed by #3259?

Well, we said back there that [your problem example in #3239](https://github.com/SciTools/iris/issues/3239#issuecomment-475674052) meant it probably *wasn't* all fixed.

So I tried that again, and that case is definitely ***not*** fixed:
```
>>> print(temp_cube.coord('time'))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/h05/itpp/git/iris/iris_main/lib/iris/coords.py", line 786, in __str__
    points = self._str_dates(self.points)
  File "/home/h05/itpp/git/iris/iris_main/lib/iris/coords.py", line 769, in _str_dates
    date_obj_array = self.units.num2date(dates_as_numbers)
  File "/opt/scitools/environments/default/2019_02_27/lib/python3.6/site-packages/cf_units/__init__.py", line 1989, in num2date
    return _num2date_to_nearest_second(time_value, cdf_utime)
  File "/opt/scitools/environments/default/2019_02_27/lib/python3.6/site-packages/cf_units/__init__.py", line 602, in _num2date_to_nearest_second
    dates = utime.num2date(time_values)
  File "cftime/_cftime.pyx", line 1211, in cftime._cftime.utime.num2date
  File "/opt/scitools/environments/default/2019_02_27/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 279, in reshape
    return _wrapfunc(a, 'reshape', newshape, order=order)
  File "/opt/scitools/environments/default/2019_02_27/lib/python3.6/site-packages/numpy/core/fromnumeric.py", line 51, in _wrapfunc
    return getattr(obj, method)(*args, **kwds)
ValueError: cannot reshape array of size 1 into shape (5,)
>>> 
```

Frankly, though, it doesn't seem to me that this problem has **anything** to do with the zero dates issue.
However it is a peculiar + mysterious problem, so I created a new ticket for it #3402
